### PR TITLE
[BB-6148] docs: add installation steps for app

### PR DESCRIPTION
**Pull request by kaustavb12**
## Description

This PR adds description and installation steps for this app.

This PR also changes the keys for circle ci cache, as there were some conflicts with the old key, which led to build failures

## Supporting information

[BB-6148](https://tasks.opencraft.com/browse/BB-6148)

## Testing instructions

1. Check the readme render in the branch's main page.
2. Check the cicle ci checks have passed

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -10,6 +10,25 @@ Overview
...
+This is done so as the new format looks far friendlier than the orginal hashed username in places like the discussion forum.
+
+Installation
+------------
+
+1. Login to LMS Shell
+2. Checkout this repository
+3. ``cd enable-retired-username-email``
+4. ``pip install -e .``
+5. This app could also be installed using `ansible-configurations <https://github.com/openedx/configuration/blob/f676c356a5424a52ebff01da7a8a7d96189f2579/playbooks/roles/edxapp/defaults/main.yml#L542>`_
```
Comment (farhaanbukhsh) [inline README.rst]: This can be split into development instruction and deployment instructions

Comment (farhaanbukhsh) [inline README.rst]: Can we include the configuration like :

```bash
EDXAPP_PRIVATE_REQUIREMENTS:
    - name: git+https://github.com/open-craft/enable-retired-username-email.git@v1.0#egg=enable-retired-username-email
      extra_args: -e
```

File: README.rst [hunk 2]
```diff
@@ -10,6 +10,25 @@ Overview
...
+
+No other configuration needed to run the app
```
Comment (farhaanbukhsh) [inline README.rst]: This is with the addition to ``Retirement Service`` so we need to enable Retirement Service in order to use this app fully.

**Review by farhaanbukhsh — CHANGES_REQUESTED** [review]
I have added few changes

File: README.rst (hunk 1 above)
Comment (kaustavb12) [reply README.rst]: Done

Comment (kaustavb12) [reply README.rst]: Done

File: README.rst (hunk 2 above)
Comment (kaustavb12) [reply README.rst]: AFAIK, the retirement service itself is enabled by default.
We however do need to configure the retirement states, which I have added to the docs.

I feel like instructions to enable the retirement cron job is beyond the scope of this README, as the user might not want to use the cron job but instead choose to trigger the retirement workflow using APIs.

**Review by farhaanbukhsh — APPROVED** [review]
👍 

- ✅ I tested the readme file
- ✅  I read through the code
- ❌  I checked for accessibility issues
- ❌  Includes documentation
- ❌  I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Pull request merged**